op-node/sources/l1_client.go

```diff
@@ -151,13 +155,20 @@ func (s *L1Client) GoOrUpdatePreFetchReceipts(ctx context.Context, l1Start uint6
 						time.Sleep(3 * time.Second)
 						continue
 					}
-					_, _, err = s.FetchReceipts(ctx, blockInfo.Hash)
-					if err != nil {
-						s.log.Warn("failed to pre-fetch receipts", "err", err)
-						time.Sleep(200 * time.Millisecond)
+					waitErr := s.preFetchReceiptsRateLimiter.Wait(ctx)
+					if waitErr != nil {
+						s.log.Warn("failed to wait pre-fetch receipts rateLimiter", "err", waitErr)
 						continue
 					}
-					s.log.Debug("pre-fetching receipts", "block", currentL1Block)
+
+					go func(ctx context.Context, blockInfo eth.L1BlockRef) {
```

Comment:
why not put the line 152 "L1BlockRefByNumber" here as well?

Reply:
Because when we reach the latest block height, L1BlockRefByNumber also has the effect of making our processing process wait for a period of time. We assume that if we parallelize the process of L1BlockRefByNumber, then when we reach the latest block height, the processing process will not stop, and it will continue to launch new goroutines to try to process blocks that have not yet been generated. 
On the other hand, the performance of the L1BlockRefByNumber interface is not so bad, and it also has its own cache, so there is no need to parallelize the processing.